lib/galaxy/webapps/galaxy/api/users.py

```diff
@@ -242,17 +242,36 @@ def delete(self, trans, id, **kwd):
         """
         if not trans.app.config.allow_user_deletion:
             raise exceptions.ConfigDoesNotAllowException('The configuration of this Galaxy instance does not allow admins to delete users.')
+        user = self.get_user(trans, id)
         purge = util.string_as_bool(kwd.get('purge', False))
         if purge:
-            raise exceptions.NotImplemented('Purge option has not been implemented yet')
-        user = self.get_user(trans, id)
-        self.user_manager.delete(user)
+            if not user.deleted:
+                raise exceptions.MessageException('Must delete the user before purging.')
+            else:
+                self.user_manager.purge(user)
+        else:
+            self.user_manager.delete(user)
         return self.user_serializer.serialize_to_view(user, view='detailed')
 
     @expose_api
     @web.require_admin
-    def undelete(self, trans, **kwd):
-        raise exceptions.NotImplemented()
+    def undelete(self, trans, id, **kwd):
+        """
+        UNDELETE /api/users/{id}
```

Comment:
DELETE here is an HTTP method/verb (https://www.restapitutorial.com/lessons/httpmethods.html), UNDELETE isn't an option. Did you use a route with POST/PUT/etc.. to test this or is it untested?